app/code/Magento/Catalog/Pricing/Price/CalculateCustomOptionCatalogRule.php

```diff
@@ -91,6 +91,7 @@ private function getGetBasePriceWithOutCatalogRules(Product $product): float
         $basePrice = null;
         foreach ($product->getPriceInfo()->getPrices() as $price) {
             if ($price instanceof BasePriceProviderInterface
+                && !$price instanceof ConfiguredPriceInterface
```

Comment:
Have you double checked that the catalog price rules are correctly applying after these changes?

Reply:
@eduard13 these changes are made because the class ConfiguredPrice calls the same function (getGetBasePriceWithOutCatalogRules() ) causing infinite recursion. This condition only prevents the function from being called again, but the rest flow is retained.

I don't think it is best solution but it gives the smallest cascade of changes in my opinion. Or i wrong ?